Translators/WZDx/main.py

```diff
@@ -75,6 +75,66 @@ def translate(wzdx_geojson):
             logging.info(f'Failed to generate TIM for feature: {feature["id"]}')
     return tims
 
+def translate(wzdx_geojson):
+    tims = {"timRwList": []}
+
+    for feature in wzdx_geojson:
+        if (len(feature["geometry"]["coordinates"]) <= 2):
+            continue
+        tim_body = {}
+        tim_body["direction"] = "I" if feature["properties"]["core_details"]["direction"].lower() in ["northbound", "eastbound"] else "D"
+        tim_body["bearing"] = get_bearing(feature)
+        tim_body["geometry"] = get_geometry(feature["geometry"]["coordinates"])
+        tim_body["route"] = feature["properties"]["core_details"]["road_names"][0].replace("_", "-")
+        tim_body["roadCode"] = feature["properties"]["core_details"]["name"].replace("_", "-")
+        tim_body["itisCodes"] = get_itis_codes(feature)
+        tim_body["action"] = ""
+        tim_body["schedStart"] = datetime.strptime(feature["properties"]["start_date"], "%Y-%m-%dT%H:%M:%SZ").strftime("%Y-%m-%d")
+        tim_body["schedEnd"] = datetime.strptime(feature["properties"]["end_date"], "%Y-%m-%dT%H:%M:%SZ").strftime("%Y-%m-%d")
+        tim_body["highway"] = feature["properties"]["core_details"]["road_names"][0].replace("_", "-")
+        tim_body["id"] = feature["properties"]["core_details"]["name"].replace("_", "-")
+        tim_body["projectKey"] = 1
+        tim_body["buffers"] = []
+        active_tim_record = active_tim(feature, tim_body)
+        if active_tim_record:
+            logging.info(f"TIM already active for record: {tim_body['id']}")
+            continue
+        tims["timRwList"].append(tim_body)
+    return tims
+
+def active_tim(feature, tim_body):
+    tim_id = tim_body["id"]
+    # if TIM has an active TIM holding record that is current & info is the same as the current TIM record, then do not update
+    active_tim_holding = query_db(f"SELECT * FROM active_tim_holding WHERE client_id LIKE '%{tim_id}%'")

```

Comment:
(praise) Will this translator have direct access to the database? That's cool!

Reply:
It will!